app/test/unit/stores/updates/update-changed-files-test.ts

```diff
@@ -118,39 +120,62 @@ describe('updateChangedFiles', () => {
       const status = createStatus({
         workingDirectory: WorkingDirectoryStatus.fromFiles(files),
       })
-      const { selectedFileIDs } = updateChangedFiles(prevState, status, false)
+      const { selection } = updateChangedFiles(prevState, status, false)
 
-      expect(selectedFileIDs).toHaveLength(1)
-      // NOTE: `updateChangedFiles` sorts the paths and `app/package.json` will
-      // appear in list before `README.md`
-      expect(selectedFileIDs[0]).toBe(files[1].id)
+      expect(selection.kind).toBe(ChangesSelectionKind.WorkingDirectory)
+
+      if (selection.kind === ChangesSelectionKind.WorkingDirectory) {
```

Comment:
:art: because of how union type checks don't flow through Jest assertions, we've previously gone through this sort of pattern to ensure assertions are always run:

```ts
expect(selection.kind).toBe(ChangesSelectionKind.WorkingDirectory)
const workingDirectorySelection = selection as ChangesWorkingDirectorySelection
const { selectedFileIDs } = workingDirectorySelection
expect(selectedFileIDs).toHaveLength(1)
expect(selectedFileIDs[0]).toBe(firstFile)
```

I'll put together a cleanup PR for this so it doesn't hold up the PR, just wanted to call this out in advance